### PR TITLE
Pdg 28 - POC du GameBoard

### PR DESCRIPTION
**Pull request by lognaume**

### CODE REVIEW

File: client/src/main/java/ch/heig/controllers/GameBoardController.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        LetterClickedCallback letterClickedCallback = new LetterClickedCallback();
+        EmptyTileClickedCallback emptyTileClickedCallback = new EmptyTileClickedCallback();
+        LetterDoubleClickedCallback letterDoubleClickedCallback = new LetterDoubleClickedCallback();
+
+        // Create the letter boards
+        // To initialize with letters, pass a collection of letters to the constructor
+        availableLettersBoard = new ReadableLettersBoard(7, letterClickedCallback);
+        availableLettersPane.getChildren().add(availableLettersBoard);
+
+        ownLettersBoard = new PlayableLettersBoard(7, letterClickedCallback, emptyTileClickedCallback, letterDoubleClickedCallback);
```
Comment (Oceanos1) [inline client/src/main/java/ch/heig/controllers/GameBoardController.java]: nombres magiques

File: client/src/main/java/ch/heig/ui/LetterTile.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.layout.StackPane;
+import javafx.scene.text.Text;
+
+import java.io.IOException;
+
+/**
+ * Class description
```
Comment (Oceanos1) [inline client/src/main/java/ch/heig/ui/LetterTile.java]: javadoc

File: client/src/main/java/ch/heig/ui/PlayableLettersBoard.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
+package ch.heig.ui;
+
+import ch.heig.exceptions.EmptyLettersBoardException;
+import javafx.event.EventHandler;
+import javafx.scene.input.MouseEvent;
+import javafx.util.Callback;
+
+/**
+ * Class description
```
Comment (Oceanos1) [inline client/src/main/java/ch/heig/ui/PlayableLettersBoard.java]: javadoc

File: client/src/main/java/ch/heig/controllers/GameBoardController.java (hunk 1 above)
Comment (lognaume) [reply client/src/main/java/ch/heig/controllers/GameBoardController.java]: Dne

**Comment by Farenjihn** [comment]
1 commit behind master, merge please

File: client/src/main/java/ch/heig/ui/LetterTile.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+ */
+public class LetterTile extends StackPane {
+    @FXML
+    private Text letterText;
+
+    @FXML
+    private Text valueText;
+
+    private char character;
+    private int value;
```
Comment (Farenjihn) [inline client/src/main/java/ch/heig/ui/LetterTile.java]: Unify style, use java wrapping instances (Integer, Boolean, etc...)

File: client/src/main/resources/stylesheet/gameBoard.css [hunk 5]
```diff
@@ -0,0 +1,18 @@
+.score {
+  -fx-font-size: 60;
+}
+
+.messageText {
+  -fx-text-fill: white;
+}
+
+.lettersBoard {
+  /* -fx-background-color: black, white;
```
Comment (Farenjihn) [inline client/src/main/resources/stylesheet/gameBoard.css]: No use in keeping this

**Comment by Oceanos1** [comment]
Y a un bug qui arrive surement a cause de la concurrence (ce bug est rare)

Stack trace

xception in thread "JavaFX Application Thread" java.lang.IndexOutOfBoundsException: Index: 8, Size: 8
	at java.util.ArrayList.rangeCheck(ArrayList.java:653)
	at java.util.ArrayList.get(ArrayList.java:429)
	at javafx.scene.Scene$MouseHandler.handleNodeRemoval(Scene.java:3575)
	at javafx.scene.Scene$MouseHandler.access$7600(Scene.java:3485)
	at javafx.scene.Scene.generateMouseExited(Scene.java:3482)
	at javafx.scene.Parent$2.onProposedChange(Parent.java:487)
	at com.sun.javafx.collections.VetoableListDecorator.clear(VetoableListDecorator.java:294)
	at ch.heig.ui.GridCell.removeLetter(GridCell.java:57)
	at ch.heig.ui.ReadableLettersBoard.removeLetter(ReadableLettersBoard.java:117)
	at ch.heig.controllers.GameBoardController.lambda$getOnKeyPressed$0(GameBoardController.java:172)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)

File: client/src/main/java/ch/heig/ui/LetterTile.java (hunk 4 above)
Comment (lognaume) [reply client/src/main/java/ch/heig/ui/LetterTile.java]: @Farenjihn I just did the exact opposite <3

Comment (lognaume) [reply client/src/main/java/ch/heig/ui/LetterTile.java]: Plz approve

**Pull request merged**